sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -1307,6 +1311,15 @@ static String getExtractDestinationUri(String extractDestinationDir) {
       return toBuilder().setNumFileShards(numFileShards).build();
     }
 
+    public Write<T> withCustomGcsTempLocation(ValueProvider<String> customGcsTempLocation) {
```

Comment:
Document these methods and explain why they might be needed

Reply:
done